### DISCUSSION TITLE
How to flash the sysex file on a STM32F411 Black Pill?

### DISCUSSION THREAD

**Discussion by zsound** [shanteacontrols/OpenDeck#35] [Q&A]
Hi, i've just bought an STM32F411 board and since one hour, i'm looking how to flash the firmware (sysex file) on it? i have searching on the open deck website and the wiki page "Firmware update via bootloader"is not very clear for me. I need more details, a complete method for flashing it (witch tools, software,etc) . Thanks you

**Comment by paradajz** [comment]
.sysex file is not meant to be flashed. It's meant to be updated via bootloader once in OpenDeck bootloader mode. Actual binary to flash STM is located in `bin/compiled` directory. Documentation for flashing is still work in progress.

**Comment by gitrimius** [comment]
Hi, last week I ordered a STM32F411 to try out Opendeck. I was looking for a description on how to flash the board..
I hope the documentation is coming soon, very curious to get started ;-) Any tips?
All this look very promising, tnx!

**Reply by paradajz** [reply]
This is a one-man person project which is developed in my spare time, and I don't have too much of it lately. Please be patient. Binaries are here: https://github.com/paradajz/OpenDeck/tree/master/bin/compiled/merged/stm32

You can find plenty of information on how to flash STM32 MCUs online until I write my own documentation.

**Reply by gitrimius** [reply]
I am so sorry if I stressed you that was not my intention... Very eager to try this out, that's all ;-) New to all this so I was unaware that I could find out online.. sorry again and tnx for the quick reply!

**Comment by gitrimius** [comment]
@zsound 
I have been successfully flashing opendeck firmware on STM32F411 yesterday following this tutorial:

https://blog.hobbycomponents.com/?p=758
Hope this helps!

**Reply by raytong** [reply]
Hi @gitrimius  ,thank you for your great info, I read the article and followed the steps , now I can run the blinking program from the Arduino IDE with STM32F401, but I still couldn't find out how to flash OpenDeck to the board, could you please give me a more detailed explanation?

**Reply by gitrimius** [reply]
I have followed this page from the Wiki:
https://github.com/paradajz/OpenDeck/wiki/Creating-custom-board-variant
btw read the full wiki, written very well and very informative...

**Comment by paradajz** [comment]
Official instructions are now available here:
https://github.com/paradajz/OpenDeck/wiki/Flashing-the-OpenDeck-firmware

**Reply by AlpesMachines** [reply]
i've successfully flashed a bare blackpill 411 thanks to the very clear instructions. the integrated switch output midi noteOn/Off C-1 velocity 127/0 through USB. If i have well understood, I can make a variant with shift registers and multiplexers, but how do i declare the used pins ? A0, A1 ... C13 ?

**Reply by paradajz** [reply]
Yes you can. See this page for info on that:
https://github.com/paradajz/OpenDeck/wiki/Creating-custom-board-variant
